staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/controller.go

```diff
@@ -201,8 +201,10 @@ func buildVersionSpecs(crd *apiextensionsv1.CustomResourceDefinition, oldSpecs m
 		if !v.Served {
 			continue
 		}
-		// Defaults are not pruned here, but before being served.
```

Comment:
@apelisse - I vaguely remember it being significant that defaults were still present here ... are they needed for SSA or something?

Reply:
Correct, we need it to find associative keys when the key is partly default-able (e.g. protocol=TCP in the port/protocol pairs)

Reply:
I'm not sure if I'm relieved tests passed with this change and that means SSA use of defaulting is still working, or concerned that pruning the defaults maybe disrupted the SSA default-dependent bits and we might be missing test coverage

Reply:
Yeah, I'd be surprised if we don't have a test for that ... Can it be that we build the SSA model before these are pruned? (I haven't looked at this PR yet).

Reply:
I'm trying to figure out the two paths metadata definitions took to get to the aggregator with defaults both pruned and not pruned...

it looks like the aggregator was already pruning defaults before aggregating: https://github.com/kubernetes/kubernetes/blob/1290d6a4c0c5cc4f758b428993d2c3760d9cf279/staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/aggregator/aggregator.go#L188

How were we getting specs with and without defaults triggering the _v2 suffixes?

Reply:
The merge bug isn't happening in the aggregator. It's happening when constructing the CRDs and merging them with the static spec directly in the apiextensions controller https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/controller.go#L230

The static spec passed in here has defaults pruned while the CRD specs do not.